compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -734,30 +746,34 @@ setReductionVectorDistributionConfig(IREE::GPU::TargetAttr target,
                         .getZExtValue();
   }
 
-  int64_t parallelSize = 1;
+  std::optional<int64_t> parallelSize = 1;
   for (int64_t dim : parallelDims) {
-    parallelSize *= bounds[dim];
+    if (ShapedType::isDynamic(bounds[dim])) {
+      parallelSize = std::nullopt;
+      break;
+    }
+    *parallelSize *= bounds[dim];
   }
 
   // Total parallel size that can fill the GPU with enough workgorups.
   // TODO: query from the target device; roughly 2x hardware compute unit.
   const int parallelThreshold = 256;
   // How many 128-bit vectors each thread should at least read.
   const int targetVectorCount = 8;
-  while (parallelSize > parallelThreshold &&
+  while (parallelSize && *parallelSize > parallelThreshold &&
          (workgroupSize / 2) % subgroupSize == 0 &&
          reductionSize / (workgroupSize * vectorSize) < targetVectorCount) {
     // Use less subgroups per workgroup..
     workgroupSize /= 2;
     // in order to host more workgroups per hardware compute unit.
-    parallelSize /= 2;
+    *parallelSize /= 2;
   }
 
   // TODO(pashu123): Currently, the threadLoads is done on the basis of
```

Comment:
Does this comment need updating?
```suggestion
  // TODO(pashu123): Currently, the vectorSize is done on the basis of
```

Reply:
I changed the var name to threadLoads (which seems more reasonable).